File: tags/astenn-1.0.1/astenn-core/src/test/java/org/lestr/astenn/test/TestCompositePersistenceDriver.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package org.lestr.astenn.test;

import org.lestr.astenn.configuration.CompositePersistenceDriver;
import org.lestr.astenn.plugin.IPersistenceDriver;
import org.lestr.astenn.configuration.RAMPersistenceDriver;
import junit.framework.TestCase;

/**
 *
 * @author PIBONNIN
 */
public class TestCompositePersistenceDriver extends TestCase {


    protected IPersistenceDriver persistenceDriver;
    protected IPersistenceDriver writeReadPersistenceDriver;
    protected IPersistenceDriver readPersistenceDriver;


    @Override
    protected void setUp() throws Exception {

        writeReadPersistenceDriver = new RAMPersistenceDriver();
        readPersistenceDriver = new RAMPersistenceDriver();
        persistenceDriver = new CompositePersistenceDriver(writeReadPersistenceDriver, readPersistenceDriver);

    }// END Method setUp


    public void test1() {

        persistenceDriver.addPluginInterface("test");

        assertTrue(persistenceDriver.getPluginInterfacesNames().iterator().hasNext());
        assertTrue(persistenceDriver.existPluginInterface("test"));
        assertTrue(writeReadPersistenceDriver.getPluginInterfacesNames().iterator().hasNext());
        assertTrue(writeReadPersistenceDriver.existPluginInterface("test"));
        assertFalse(readPersistenceDriver.getPluginInterfacesNames().iterator().hasNext());
        assertFalse(readPersistenceDriver.existPluginInterface("test"));

        persistenceDriver.addPluginImplementation("test", "local:test");

        assertTrue(persistenceDriver.getPluginImplementationsAddresses("test").iterator().hasNext());
        assertTrue(persistenceDriver.existPluginImplementation("test", "local:test"));
        assertTrue(writeReadPersistenceDriver.existPluginImplementation("test", "local:test"));
        assertFalse(readPersistenceDriver.existPluginImplementation("test", "local:test"));

        persistenceDriver.removePluginImplementation("test", "local:test");

        assertTrue(persistenceDriver.existPluginInterface("test"));
        assertFalse(persistenceDriver.existPluginImplementation("test", "local:test"));
        assertTrue(writeReadPersistenceDriver.existPluginInterface("test"));
        assertFalse(writeReadPersistenceDriver.existPluginImplementation("test", "local:test"));
        assertFalse(readPersistenceDriver.existPluginInterface("test"));
        assertFalse(readPersistenceDriver.existPluginImplementation("test", "local:test"));

        persistenceDriver.removePluginInterface("test");

        assertFalse(persistenceDriver.existPluginInterface("test"));
        assertFalse(writeReadPersistenceDriver.existPluginInterface("test"));
        assertFalse(readPersistenceDriver.existPluginInterface("test"));

        readPersistenceDriver.addPluginInterface("test");
        readPersistenceDriver.addPluginImplementation("test", "local:test");

        assertTrue(persistenceDriver.existPluginInterface("test"));
        assertFalse(writeReadPersistenceDriver.existPluginInterface("test"));
        assertTrue(readPersistenceDriver.existPluginInterface("test"));
        assertTrue(persistenceDriver.existPluginImplementation("test", "local:test"));
        assertFalse(writeReadPersistenceDriver.existPluginImplementation("test", "local:test"));
        assertTrue(readPersistenceDriver.existPluginImplementation("test", "local:test"));

        readPersistenceDriver.removePluginImplementation("test", "local:test");

        assertFalse(persistenceDriver.existPluginImplementation("test", "local:test"));
        assertFalse(writeReadPersistenceDriver.existPluginImplementation("test", "local:test"));
        assertFalse(readPersistenceDriver.existPluginImplementation("test", "local:test"));
        
        readPersistenceDriver.removePluginInterface("test");

        assertFalse(persistenceDriver.existPluginInterface("test"));
        assertFalse(writeReadPersistenceDriver.existPluginInterface("test"));
        assertFalse(readPersistenceDriver.existPluginInterface("test"));

    }// END Method test1


}// END Class TestCompositePersistenceDriver

File: trunk/astenn-core/src/test/csharp/src/Org/Lestr/Astenn/Test/TestCompositePersistenceDriver.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Linq;

using NUnit.Framework;

using Org.Lestr.Astenn.Plugin;
using Org.Lestr.Astenn.Configuration;

namespace Org.Lestr.Astenn.Test
{

    [TestFixture]
    public class TestCompositePersistenceDriver
    {

        protected IPersistenceDriver persistenceDriver;
        protected IPersistenceDriver writeReadPersistenceDriver;
        protected IPersistenceDriver readPersistenceDriver;


        [SetUp]
        public void setUp() {

            writeReadPersistenceDriver = new RAMPersistenceDriver();
            readPersistenceDriver = new RAMPersistenceDriver();
            persistenceDriver = new CompositePersistenceDriver(writeReadPersistenceDriver, new IPersistenceDriver[]{ readPersistenceDriver });

        }// END Method setUp


        [Test]
        public void test1()
        {

            persistenceDriver.AddPluginInterface("test");

            Assert.True(persistenceDriver.GetPluginInterfacesNames().Count() > 0);
            Assert.True(persistenceDriver.ExistPluginInterface("test"));
            Assert.True(writeReadPersistenceDriver.GetPluginInterfacesNames().Count() > 0);
            Assert.True(writeReadPersistenceDriver.ExistPluginInterface("test"));
            Assert.False(readPersistenceDriver.GetPluginInterfacesNames().Count() > 0);
            Assert.False(readPersistenceDriver.ExistPluginInterface("test"));

            persistenceDriver.AddPluginImplementation("test", "local:test");

            Assert.True(persistenceDriver.GetPluginImplementationsAddresses("test").Count() > 0);
            Assert.True(persistenceDriver.ExistPluginImplementation("test", "local:test"));
            Assert.True(writeReadPersistenceDriver.ExistPluginImplementation("test", "local:test"));
            Assert.False(readPersistenceDriver.ExistPluginImplementation("test", "local:test"));

            persistenceDriver.RemovePluginImplementation("test", "local:test");

            Assert.True(persistenceDriver.ExistPluginInterface("test"));
            Assert.False(persistenceDriver.ExistPluginImplementation("test", "local:test"));
            Assert.True(writeReadPersistenceDriver.ExistPluginInterface("test"));
            Assert.False(writeReadPersistenceDriver.ExistPluginImplementation("test", "local:test"));
            Assert.False(readPersistenceDriver.ExistPluginInterface("test"));
            Assert.False(readPersistenceDriver.ExistPluginImplementation("test", "local:test"));

            persistenceDriver.RemovePluginInterface("test");

            Assert.False(persistenceDriver.ExistPluginInterface("test"));
            Assert.False(writeReadPersistenceDriver.ExistPluginInterface("test"));
            Assert.False(readPersistenceDriver.ExistPluginInterface("test"));

            readPersistenceDriver.AddPluginInterface("test");
            readPersistenceDriver.AddPluginImplementation("test", "local:test");

            Assert.True(persistenceDriver.ExistPluginInterface("test"));
            Assert.False(writeReadPersistenceDriver.ExistPluginInterface("test"));
            Assert.True(readPersistenceDriver.ExistPluginInterface("test"));
            Assert.True(persistenceDriver.ExistPluginImplementation("test", "local:test"));
            Assert.False(writeReadPersistenceDriver.ExistPluginImplementation("test", "local:test"));
            Assert.True(readPersistenceDriver.ExistPluginImplementation("test", "local:test"));

            readPersistenceDriver.RemovePluginImplementation("test", "local:test");

            Assert.False(persistenceDriver.ExistPluginImplementation("test", "local:test"));
            Assert.False(writeReadPersistenceDriver.ExistPluginImplementation("test", "local:test"));
            Assert.False(readPersistenceDriver.ExistPluginImplementation("test", "local:test"));

            readPersistenceDriver.RemovePluginInterface("test");

            Assert.False(persistenceDriver.ExistPluginInterface("test"));
            Assert.False(writeReadPersistenceDriver.ExistPluginInterface("test"));
            Assert.False(readPersistenceDriver.ExistPluginInterface("test"));
            
        }// END Method test1


    }// END Class TestCompositePersistenceDriver


}// END Namespace Org.Lestr.Astenn.Test